### PR TITLE
ansible role added

### PR DESCRIPTION
**Pull request by mhkarimi1383**
using ansible you can automate installation process and make it cleaner and faster  
this playbook will install docker
then install uptime kuma using docker
and install and configure nginx with ssl

### CODE REVIEW

**Comment by louislam** [comment]
> Any progress You will accept it or not?

It could be accept with following constraints:

- It will be added in a secondary branch as a unofficial installation method. It will also be added to the wiki too.
  Similar to Kubernetes: https://github.com/louislam/uptime-kuma/tree/k8s-unofficial/kubernetes
- It needs one more reviewer to test and to confirm the ansible script.

**Comment by mhkarimi1383** [comment]
> > Any progress You will accept it or not?
> 
> It could be accept with following constraints:
> 
> * It will be added in a secondary branch as a unofficial installation method. It will also be added to the wiki too.
>   Similar to Kubernetes: https://github.com/louislam/uptime-kuma/tree/k8s-unofficial/kubernetes
> * It needs one more reviewer to test and to confirm the ansible script.

Thanks
I think should wait for another user I think to approve it.

**Comment by louislam** [comment]
> I should still wait?

An extra tester is still needed.

**Comment by gaby** [comment]
I don't think this role should install Docker, Docker-Compose, etc on a host. There's plenty of official roles for doing that kind of stuff, no need to reinvent the wheel. This role should ONLY cover `Uptime-Kuma` provisioning.

File: ansible/roles/uptime-kuma/templates/docker-compose.yml
Comment (gaby) [reply ansible/roles/uptime-kuma/templates/docker-compose.yml:10]: Still valid comment

File: ansible/roles/nginx/templates/docker-compose.yml
Comment (gaby) [reply ansible/roles/nginx/templates/docker-compose.yml:4]: This shouldn't be using HOST network. Use port-binding to port 80 and 443

File: ansible/roles/uptime-kuma/templates/docker-compose.yml
Comment (gaby) [inline ansible/roles/uptime-kuma/templates/docker-compose.yml:6]: There's not need to use `ports` here, only NGINX should be binding ports to the host. Instead use `expose`, which only exposes the port to the Docker network.

```
expose:
  - 3001
```

File: ansible/roles/nginx/templates/nginx.conf
Comment (gaby) [inline ansible/roles/nginx/templates/nginx.conf:82]: This should be: 

```
proxy_pass	http://uptime-kuma:3001;
```

Since the service gets a hostname assigned by Docker Network.

File: ansible/roles/nginx/templates/docker-compose.yml
Comment (gaby) [inline ansible/roles/nginx/templates/docker-compose.yml:3]: Why is NGINX by itself on a separate Docker-Compose file?

The current NGINX is only reverse-proxying Uptime-Kuma, so they should be both on the same file that way you don't have to manage multiple Docker Networks.

File: ansible/roles/nginx/templates/nginx.conf
Comment (gaby) [inline ansible/roles/nginx/templates/nginx.conf:70]: Suggest creating a folder: 

`/var/log/nginx/uptime-kuma`

Which will container 

```
access.log
error.log

```

File: ansible/roles/uptime-kuma/templates/docker-compose.yml
Comment (gaby) [inline ansible/roles/uptime-kuma/templates/docker-compose.yml:3]: This whole compose file should be merge with the NGINX one. No need to run 2 compose instances if NGINX is only managing traffic for `uptime-kuma`

File: ansible/playbook.yml
Comment (gaby) [inline ansible/playbook.yml:7]: docker-compose 2.0.1 is not installable via PIP. They migrated from Python to Golang for v2.x.x

**Review by gaby — CHANGES_REQUESTED** [review]

**Comment by gaby** [comment]
@louislam @mhkarimi1383 Added comment.

**Comment by gaby** [comment]
@mhkarimi1383 

Example docker-compose.yml

```
version: '3'

services:
    nginx:
        image: nginx:stable
        container_name: nginx
        restart: always
        networks:
			- uptime-kuma
        ports:
            - 443:443
            - 80:80
        volumes:
            - /compose/volumes/nginx/:/etc/nginx/
            
    uptime-kuma:
        image: louislam/uptime-kuma:1-debian
        container_name: uptime-kuma
        restart: always
        networks:
			- uptime-kuma
        expose:
        	- 3001
        volumes:
            - /compose/volumes/uptime-kuma:/app/data

networks:
  uptime-kuma: 
```

**Comment by mhkarimi1383** [comment]
I will add your suggestions...
 thanks

**Comment by mhkarimi1383** [comment]
but for docker tag I'm not sure.....

**Comment by mhkarimi1383** [comment]
@gaby You suggestions Are applied with some improvements

**Review by gaby — APPROVED** [review]

**Comment by gaby** [comment]
👍🏻 Looks good to me

**Comment by mhkarimi1383** [comment]
Still waiting :)

**Comment by mhkarimi1383** [comment]
I'm waiting for this merge :)

**Comment by louislam** [comment]
Have been merged into `ansible-unofficial` branch:
https://github.com/louislam/uptime-kuma/tree/ansible-unofficial/ansible

Also added in "How to Install":
https://github.com/louislam/uptime-kuma/wiki/%F0%9F%94%A7-How-to-Install

Thank you so much.
Feel free to make a new pull request to `ansible-unofficial` branch if you need to update the ansible config in the future.

**Comment by mhkarimi1383** [comment]
Thanks :)

**Pull request closed without merging**
